src/ImageScaler.java

```diff
@@ -0,0 +1,169 @@
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.awt.image.WritableRaster;
+import java.io.File;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public class ImageScaler {
+    private static BufferedImage cutImage(BufferedImage image) {
+        WritableRaster writableRaster = image.getAlphaRaster();
+        int width = writableRaster.getWidth(), height = writableRaster.getHeight();
+        int left = 0, top = 0;
+        int right = width - 1, minRight = width - 1;
+        int bottom = height - 1, minBottom = height - 1;
+
+        boolean foundTop = false;
+        while (top < bottom && !foundTop) {
+            int x = 0;
+            while (x < width) {
+                if (writableRaster.getSample(x, top, 0) != 0) {

```

Comment:
To działa? Bo u mnie getSample na tym AlphaRasterze zawsze zwraca 255 dla wszystkich pixeli oraz bands. Może gdzieś popełniłem "czeski błąd", ale czy nie prościej byłoby po prostu użyć getRGB z BufferedImage? -1 to wtedy biały.